src/Tasks/Microsoft.NET.Build.Tasks/PrepareForReadyToRunCompilation.cs

```diff
@@ -305,6 +305,22 @@ private bool ReferencesWinMD(MetadataReader mdReader)
             return false;
         }
 
+        private bool HasILCode(PEReader peReader, MetadataReader mdReader)
+        {
+            foreach (var methoddefHandle in mdReader.MethodDefinitions)
+            {
+                MethodDefinition methodDef = mdReader.GetMethodDefinition(methoddefHandle);
+                if (methodDef.RelativeVirtualAddress > 0)
+                {
+                    MethodBodyBlock methodBody = PEReaderExtensions.GetMethodBody(peReader, methodDef.RelativeVirtualAddress);
```

Comment:
nit, this is an extension method, so you can call it as peReader.GetMethodBody(methodDef.RelativeVirtualAddress).

@tmat is there a faster way to determine if there is an IL than to check every method def?

Reply:
> peReader.GetMethodBody(methodDef.RelativeVirtualAddress)

Wouldn't this throw if the RVA is 0?

Reply:
This is just a syntactic transformation. (hence "nit") These two compile to the exact same IL:

PEReaderExtensions.GetMethodBody(peReader, methodDef.RelativeVirtualAddress);
peReader.GetMethodBody(methodDef.RelativeVirtualAddress);

It's just how the `this PEReader` parameter works here: https://source.dot.net/#System.Reflection.Metadata/System/Reflection/Metadata/PEReaderExtensions.cs,25

So if you can do the first one, you can write it in the second form.

Reply:
I think it would be sufficient to check RVA != 0 for all method defs.
Unfortunately, there is no "IL stream" directory entry in the headers that would specify where to find the IL stream.

---
In reply to: [273174990](https://github.com/dotnet/sdk/pull/3099#discussion_r273174990) [](ancestors = 273174990)

Reply:
Thanks folks. I'll change this to just checking the RVA